### ISSUE TITLE
`props` option of `no-param-reassign` cannot handle es6 destructuring

### ISSUE THREAD

**Issue by sodatea** [eslint/eslint#4635]
## ESLint version

v1.10.3

## test.js

```javascript
/* eslint no-param-reassign: [2, { props: false }] */
const jschardet = require('jschardet');

function convert(buf, { from, to = 'utf-8', minConfidence = 0.96 }) {
    if (!from) {
        const detected = jschardet.detect(buf);
        from = detected.confidence > minConfidence ? detected.encoding : 'utf-8';
    }
    // ...
}
```

## Command and output

```
eslint ./test.js --env es6 --no-eslintrc                                                                                         130 ↵

/Users/soda/Tests/eslint/test.js
  6:9  error  Assignment to function parameter 'from'  no-param-reassign

✖ 1 problem (1 error, 0 warnings)
```

## Expected

Since `from` is a property of the second parameter, there should be no errors.

**Comment by nzakas** [comment]
Working as expected. `from` is a local binding that contains the same value as a property but I'd not a property itself.
